epochStart/metachain/sovereignValidators_test.go

```diff
@@ -0,0 +1,191 @@
+				Type:            block.StateBlock,
+				TxHashes: [][]byte{
+					[]byte("txHash3"),
+				},
+			},
+		},
+	}
+	marshalledData := svic.CreateMarshalledData(body)
+	require.Equal(t, 1, len(marshalledData))
+	require.Equal(t, 2, len(marshalledData[common.ValidatorInfoTopic]))
+	require.Equal(t, marshalledSVI1, marshalledData[common.ValidatorInfoTopic][0])
+	require.Equal(t, marshalledSVI2, marshalledData[common.ValidatorInfoTopic][1])
+
+	cachedData1, found := shardDataCacher.SearchFirstData(txHash1)
+	require.True(t, found)
+	require.Equal(t, svi1, cachedData1)
+
+	cachedData2, found := shardDataCacher.SearchFirstData(txHash2)
+	require.True(t, found)
+	require.Equal(t, svi2, cachedData2)
+}
+
+func TestSovereignValidatorInfoCreator_CreateMarshalledDataErrorCases(t *testing.T) {
+
+	t.Run("CreateMarshalledData should return nil body is nil", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := createMockEpochValidatorInfoCreatorsArguments()
+		vic, _ := NewValidatorInfoCreator(arguments)
+		svic, _ := NewSovereignValidatorInfoCreator(vic)
+
+		marshalledData := svic.CreateMarshalledData(nil)
+		require.Nil(t, marshalledData)
+	})
+
+	t.Run("CreateMarshalledData should return empty slice when tx hash does not exist in validator info cacher", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := createMockEpochValidatorInfoCreatorsArguments()
+		localErr := errors.New("local error")
+		arguments.DataPool = &dataRetrieverMock.PoolsHolderStub{
+			CurrEpochValidatorInfoCalled: func() dataRetriever.ValidatorInfoCacher {
+				return &vics.ValidatorInfoCacherStub{
+					GetValidatorInfoCalled: func(validatorInfoHash []byte) (*state.ShardValidatorInfo, error) {
+						return nil, localErr
+					},
+				}
+			},
+		}
+		wasMarshallCalled := false
+		arguments.Marshalizer = &testscommon.MarshallerStub{
+			MarshalCalled: func(obj interface{}) ([]byte, error) {
+				wasMarshallCalled = true
+				return nil, nil
+			},
+		}
+
+		vic, _ := NewValidatorInfoCreator(arguments)
+		svic, _ := NewSovereignValidatorInfoCreator(vic)
+
+		body := createMockBlockBody(core.SovereignChainShardId, core.SovereignChainShardId, block.PeerBlock)
+		marshalledData := svic.CreateMarshalledData(body)
+		require.Empty(t, marshalledData)
+		require.False(t, wasMarshallCalled)
+	})
+
+	t.Run("CreateMarshalledData should return empty slice when marhsall fails", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := createMockEpochValidatorInfoCreatorsArguments()
+		arguments.DataPool = &dataRetrieverMock.PoolsHolderStub{
+			CurrEpochValidatorInfoCalled: func() dataRetriever.ValidatorInfoCacher {
+				return &vics.ValidatorInfoCacherStub{
+					GetValidatorInfoCalled: func(validatorInfoHash []byte) (*state.ShardValidatorInfo, error) {
+						return &state.ShardValidatorInfo{}, nil
+					},
+				}
+			},
+		}
+		localErr := errors.New("local error")
+
+		arguments.Marshalizer = &testscommon.MarshallerStub{
+			MarshalCalled: func(obj interface{}) ([]byte, error) {
+
```

Comment:
Maybe add another `wasMarshallCalled = true` here and `require.True` at line 188.
Or delete this empty line.

Reply:
No need, I already check `shardDataCacher.Keys` to be empty

Reply:
`shardDataCacher.Keys` is always empty and `ValidatorsInfoCalled` is never called in this test.